src/core/libmaven/mzSample.cpp

```diff
@@ -1257,16 +1257,19 @@ EIC *mzSample::getEIC(float precursorMz, float collisionEnergy, float productMz,
 		}
 
 		int lastIndex = e->rt.size()-1;
+		//if rt is already present save the higher intensity for that rt
+		//this can happen when there are multiple product m/z for the same precursor
 		if (lastIndex != -1 && scan->rt == e->rt[lastIndex]) {
 			if (eicIntensity <= e->intensity[lastIndex]) 
```

Comment:
std::find() can be used to get the index. e.g  vector.begin() - std::find(vector.begin(), vector.end(),  value)
This can stored in the previous step, where we try to find the  rt